src/main/scala/scorex/core/network/NodeViewSynchronizer.scala

```diff
@@ -224,23 +234,25 @@ class NodeViewSynchronizer[TX <: Transaction,
       val modifiers = data._2
 
       log.info(s"Got modifiers of type $typeId from remote connected peer: $remote")
-      log.trace(s"Received modifier ids ${data._2.keySet.map(encoder.encode).mkString(",")}")
+      log.trace(s"Received modifier ids ${data._2.keySet.map(id => encoder.encode(id)).mkString(",")}")
 
-      modifiers.foreach { case (id, _) => deliveryTracker.onReceive(typeId, id, remote) }
+      modifiers.foreach { case (id, _) =>
+        statusKeeper.received(id)
+        deliveryTracker.onReceive(typeId, id, remote)
+      }
 
       val (spam, fm) = modifiers.partition { case (id, _) => deliveryTracker.isSpam(id) }
 
       if (spam.nonEmpty) {
         log.info(s"Spam attempt: peer $remote has sent a non-requested modifiers of type $typeId with ids" +
-          s": ${spam.keys.map(encoder.encode)}")
+          s": ${spam.keys.map(id => encoder.encode(id))}")
```

Comment:
Do we need `encoder.encode` here?

Reply:
I think it might be useful if we'll decide to switch from `Base16` in `ModifierId` to something else, or if desired enocoding will be different from `Base16`